### PR TITLE
[semver:patch] chore: condition checkout for each platform

### PR DESCRIPTION
**Pull request by peternhale**
@W-9167855@

### CODE REVIEW

File: src/jobs/test-nut.yml [hunk 1]
```diff
@@ -45,12 +49,27 @@ steps:
...
-        not:
-          equal: ['', <<parameters.repo_tag>>]
+        and:
+          - equal: ['windows', <<parameters.os>>]
+          - not: <<parameters.repo_tag>>
+      steps:
+        - run:
+            name: Switch to tag v<<parameters.repo_tag>> or  <<parameters.npm_module_name>>@<<parameters.repo_tag>>
+            command: |
+              git checkout v<<parameters.repo_tag>>
```
Comment (mshanemc) [inline src/jobs/test-nut.yml:59]: doesn't line 54 mean there is no repo_tag?  How can it be used here?

File: src/jobs/test-nut.yml [hunk 2]
```diff
@@ -45,12 +49,27 @@ steps:
...
+              if (!$?)
+              {
+                git checkout <<parameters.npm_module_name>>@<<parameters.repo_tag>>
+              }
+  - when:
+      condition:
+        and:
+          - equal: ['linux', <<parameters.os>>]
+          - not: <<parameters.repo_tag>>
```
Comment (mshanemc) [inline src/jobs/test-nut.yml:68]: same comment.  It made more sense to me when it was 
```
not:
          equal: ['', <<parameters.repo_tag>>]
```

File: src/jobs/test-nut.yml (hunk 1 above)
Comment (peternhale) [reply src/jobs/test-nut.yml:59]: @mshanemc Good catch!

File: src/jobs/test-nut.yml (hunk 2 above)
Comment (peternhale) [reply src/jobs/test-nut.yml:68]: @mshanemc never could get that pattern to get through the pack. I'll fiddle some more.

**Review by mshanemc — APPROVED** [review]

**Pull request merged**